internal/api/routes.go

```diff
@@ -31,3 +42,33 @@ func GetRoutes() *[]*core.Route {
 
 	return &routes
 }
+
+func Describe() *APIDescription {
+	routes := GetRoutes()
+
+	endpointMap := make(map[string]EndpointDescription)
+	for _, route := range *routes {
+		apiRoute, ok := route.(*core.APIRoute)
+		if ok {
+			endpointMap[apiRoute.GetBasePath()] = EndpointDescription{
+				Description:  apiRoute.GetDescription(),
+				RequestType:  apiRoute.Request.String(),
+				ResponseType: apiRoute.Response.String(),
+			}
+		}
+	}
+
+	return &APIDescription{
+		Endpoints: map[string]map[string]EndpointDescription{
+			"endpoints": endpointMap,
+		},
+	}
+}
+
+func DescribeToJSON(description *APIDescription) (string, error) {
```

Comment:
This function doesn't seem necessary.
Also, we usually don't want to let library functions (in this case just not top-level packages) do our formatting for us.